core/src/main/java/org/bitcoinj/core/Peer.java

```diff
@@ -1437,19 +1436,21 @@ private void blockChainDownloadLocked(Sha256Hash toHash) {
         if (log.isDebugEnabled())
             log.debug("{}: blockChainDownloadLocked({}) current head = {}",
                     this, toHash, chainHead.getHeader().getHashAsString());
+        List<Sha256Hash> hashes = new ArrayList<>(100);
         StoredBlock cursor = chainHead;
         for (int i = 100; cursor != null && i > 0; i--) {
-            blockLocator = blockLocator.add(cursor.getHeader().getHash());
+            hashes.add(cursor.getHeader().getHash());
             try {
                 cursor = cursor.getPrev(store);
             } catch (BlockStoreException e) {
                 log.error("Failed to walk the block chain whilst constructing a locator");
                 throw new RuntimeException(e);
             }
         }
-        // Only add the locator if we didn't already do so. If the chain is < 50 blocks we already reached it.
+        // Only add genesis hash to locator if we didn't already do so. If the chain is < 100 blocks we already reached it.
```

Comment:
Is this comment change related to the change of BlockLocator construction? It seems to me it's unrelated.

Reply:
Unrelated. I just noticed it was wrong and fixed it. Do you want that in a separate commit?

Reply:
See PR #3151